src/mock-doc/node.ts

```diff
@@ -1062,18 +1069,21 @@ export class MockHTMLElement extends MockElement {
   }
 
   override get tagName() {
-    return this.nodeName;
+    return this.nodeName ?? '';
   }
   override set tagName(value: string) {
     this.nodeName = value;
   }
 
-  override get attributes() {
+  override get attributes(): MockAttributeMap {
     if (this.__attributeMap == null) {
-      this.__attributeMap = createAttributeProxy(true);
+      const attrMap = createAttributeProxy(true);
+      this.__attributeMap = attrMap;
+      return attrMap;
```

Comment:
same comment as earlier, we may be able to simplify this a bit:
```suggestion
      this.__attributeMap = createAttributeProxy(true);
```

Reply:
Disregard, I missed the comments inline somehow 🤔